python/lib/python_port.c

```diff
@@ -99,12 +108,14 @@ PyObject* py_port_set(PyObject* self, PyObject* args) {
 
     if (val) {
         LF_PRINT_DEBUG("Setting value %p.", val);
+        Py_INCREF(val);
         python_count_decrement(port->value);
        
         lf_token_t* token = lf_new_token((void*)port, val, 1);
-        lf_set_destructor(port, python_count_decrement);
+        lf_set_destructor(port, output_port_destructor);
         lf_set_token(port, token);
-
+        Py_INCREF(val);
```

Comment:
Why is `Py_INCREF(val)` written twice in this commit?

If the "right" destructor decrements the output by one, then probably we should preserve the invariant that the portion of the reference count that is created by our runtime is always either zero or one, depending on whether any part of our runtime has a reference to the memory. So, we should never increment the reference count twice in a row nor decrement it twice in a row.